components/HowToPanel.tsx

```diff
@@ -44,13 +46,9 @@ const HowToPanel: React.FC<HowToPanelProps> = ({
   }
 
   useOnClickOrEsc(ref, () => close())
-  let queryString
-  if (typeof window !== 'undefined') {
-    queryString = window.location.search
-  }
-  const urlParams: URLSearchParams = new URLSearchParams(queryString)
-  const urlIconName: string | null = urlParams.get('iconName')
-  const urlTagName: string | null = urlParams.get('tagName')
+
+  const urlTagName = router.query.tagName as string
+  const urlIconName = router.query.iconName as string
```

Comment:
why was the snippet updated? How can we sure that query would always contain the `tagName`, and the `iconName`

Reply:
My comment was to accommodate the types part. The suggestion was to make it `const urlTagName : ?string = urlParams.get('iconName')`.

Reply:
When you pointed at this section of code, I noticed that it was using the old way of fetching URL parameters. Using the router to fetch URL params is the preferred way in Next.js. I updated it to use router instead. 
URL doesn't contain `tagName` and `iconName` always, in that case, the values are undefined. We are putting checks in place to ensure that these values are defined where we need them.
I have typecasted the value to string because by default the value can have string[] and undefined types as well but in our use case it's always a string.